src/push/hms.rs

```diff
@@ -32,32 +34,45 @@ use crate::{
     push::{HmsToken, ThreemaPayload},
 };
 
-enum EndpointType {
-    Login,
-    Push,
+pub struct HmsStateConfig {
+    login_endpoint: Cow<'static, str>,
+    push_endpoint: Cow<'static, str>,
 }
 
-// Server endpoints
-#[cfg(not(test))]
-fn hms_endpoint(endpoint_type: EndpointType) -> &'static str {
-    match endpoint_type {
-        EndpointType::Login => "https://oauth-login.cloud.huawei.com",
-        EndpointType::Push => "https://push-api.cloud.huawei.com",
+type SharedHmsConfig = Arc<HmsStateConfig>;
+
+impl HmsStateConfig {
+    pub fn new() -> SharedHmsConfig {
```

Comment:
The `FcmStateConfig` has a `new_shared` method, while here we just use `new`. Maybe the two could be made consistent? (I think `new_shared` is pretty good naming, if the returned type is wrapped in an Arc.)

Reply:
Renamed the method for consistency